src/public/media.ts

```diff
@@ -77,3 +92,268 @@ export function captureImage(callback: (error: SdkError, files: File[]) => void)
+   * Default value is true
+   */
+  textSticker?: boolean;
+
+  /**
+   * Optional; indicate if image filtering mode is enabled on the selected image
+   * Default value is false
+   */
+  enableFilter?: boolean;
+}
+
+/**
+ * The modes in which camera can be launched in select Media API
+ */
+export const enum Mode {
+  Photo = 1,
+  Document = 2,
+  Whiteboard = 3,
+  BusinessCard = 4,
+}
+
+/**
+ * Specifies the image source
+ */
+export const enum Source {
+  Camera = 1,
+  Gallery = 2,
+}
+
+/**
+ * Specifies the type of Media
+ */
+export const enum MediaType {
+  Image = 1,
+  Video = 2,
+  ImageOrVideo = 3,
+  Audio = 4,
+}
+
+/**
+ * Input for view images API
+ */
+export interface ImageUri {
+  value: string;
+  type: ImageUriType;
+}
+
+/**
+ * ID contains a mapping for content uri on platform's side, URL is generic
+ */
+export const enum ImageUriType {
+  ID = 1,
+  URL = 2,
+}
+
+/**
+ * Media chunks an output of getMedia API from platform
+ */
+export interface MediaChunk {
+  /**
+   * Base 64 data for the requested uri
+   */
+  chunk: string;
+
+  /**
+   * chunk sequence number​
+   */
+  chunkSequence: number;
+}
+
+/**
+ * Output of getMedia API from platform
+ */
+interface MediaResult {
+  /**
+   * error encountered in getMedia API
+   */
+  error: SdkError;
+
+  /**
+   * Media chunk which will be assemebled and converted into a blob
+   */
+  mediaChunk: MediaChunk;
+}
+
+/**
+ * Helper object to assembled media chunks
+ */
+export interface AssembleAttachment {
+  sequence: number;
+  file: Blob;
+}
+
+/**
+ * Helper class for assembling media
+ */
+interface MediaHelper {
+  mediaMimeType: string;
+  assembleAttachment: AssembleAttachment[];
+}
+
+/**
+ * Select an attachment using camera/gallery
+ * @param mediaInputs The input params to customize the media to be selected
+ * @param callback The callback to invoke after fetching the media
+ */
+export function selectMedia(mediaInputs: MediaInputs, callback: (error: SdkError, attachments: Media[]) => void): void {
+  if (!callback) {
+    throw new Error('[select Media] Callback cannot be null');
+  }
+  ensureInitialized(FrameContexts.content, FrameContexts.task);
+  if (!isAPISupportedByPlatform(mediaAPIVersion)) {
+    const oldPlatformError: SdkError = { errorCode: ErrorCode.OLD_PLATFORM };
+    callback(oldPlatformError, null);
+    return;
+  } else if (!validateSelectMediaInputs(mediaInputs)) {
+    const invalidInput: SdkError = { errorCode: ErrorCode.INVALID_ARGUMENTS };
+    callback(invalidInput, null);
+    return;
+  }
+
+  const params = [mediaInputs];
+  const messageId = sendMessageRequestToParent('selectMedia', params);
+  GlobalVars.callbacks[messageId] = callback;
+}
+
+/**
+ * View images using native image viewer
+ * @param uriList urilist of images to be viewed - can be content uri or server url. supports upto 10 Images in one go
+ * @param result returns back error if encountered, there will be no callback in case of success
+ */
```

Comment:
same here support success and failure in callback explicit.

Reply:
We don't need a success callback here. Default case is assumed as success.